website/source/docs/builders/googlecompute.html.md

```diff
@@ -213,7 +213,10 @@ builder.
     instance.
 
 -   `network` (string) - The Google Compute network id or URL to use for the
-    launched instance. Defaults to `"default"`.
+    launched instance. Defaults to `"default"`. If the value is not a URL, it
+    will be interpolated to `projects/((network_project_id))/global/networks/((network))`.
```

Comment:
I'm not sure this default is in the format we like it. Normally I think we just capitalise the variable like `projects/NETWORK_PROJECT_ID/global/networks/NETWORK` ([example](https://www.packer.io/docs/builders/virtualbox-iso.html#output_directory)).

@mwhooker @SwampDragons any guidance?

Reply:
pulling other examples from existing docs..

in AWS we do stuff like

> ami_description (string) - The description to set for the resulting AMI(s). By default this description is empty. This is a template engine where the SourceAMI variable is replaced with the source AMI ID and BuildRegion variable is replaced with the value of region.

but that doesn't have a default value.

In docker, we show essentially the go template

> run_command (array of strings) - An array of arguments to pass to docker run in order to run the container. By default this is set to ["-d", "-i", "-t", "{{.Image}}", "/bin/bash"]. As you can see, you have a couple template variables to customize, as well.

And reinforcing that is the shell provisioner

> execute_command (string) - The command to use to execute the script. By default this is chmod +x {{ .Path }}; {{ .Vars }} {{ .Path }}. The value of this is treated as configuration template. There are two available variables: Path, which is the path to the script to run, and Vars, which is the list of environment_vars, if configured.

Would it make sense to turn network/region/etc into template variables? It seems like odd behavior to only get those special values by leaving it as the default, though I don't really know if they would make sense outside of the default.

I'd say if a user would ever want to construct a network with the above variables, make network a template and document them as `{{.NetworkProjectID}}`, etc. and if it doesn't make sense to do that just document them like we do for `output_directory` in @rickard-von-essen 's link